MediaSession.cpp

```diff
@@ -33,12 +33,22 @@ namespace CDMi {
 
 // The default location of CDM DRM store.
 // /tmp/drmstore.dat
+#ifdef PR_3_3
+const DRM_WCHAR g_rgwchCDMDrmStoreName[] = {DRM_WCHAR_CAST('/'), DRM_WCHAR_CAST('t'), DRM_WCHAR_CAST('m'), DRM_WCHAR_CAST('p'), DRM_WCHAR_CAST('/'),
+                                            DRM_WCHAR_CAST('d'), DRM_WCHAR_CAST('r'), DRM_WCHAR_CAST('m'), DRM_WCHAR_CAST('s'), DRM_WCHAR_CAST('t'),
+                                            DRM_WCHAR_CAST('o'), DRM_WCHAR_CAST('r'), DRM_WCHAR_CAST('e'), DRM_WCHAR_CAST('.'), DRM_WCHAR_CAST('d'),
+                                            DRM_WCHAR_CAST('a'), DRM_WCHAR_CAST('t'), DRM_WCHAR_CAST('\0')};
+
+const DRM_CONST_STRING g_dstrCDMDrmStoreName = DRM_CREATE_DRM_STRING(g_rgwchCDMDrmStoreName);
```

Comment:
This wasn't really what I had in mind. Rather than adding lots more additional #ifdef PR_3_3 throughout the code which makes it hard to read. I envisaged one place at the top which had the contents of your previous patch on internal Gerrit server e.g.

#ifdef PR_3_3
// Some macro names have changed name in the latest MS porting kit. Once support
// is no longer required for older porting kit releases, these mappings can be removed
// and the code changed to use the new names.
#define WCHAR_CAST(c) DRM_WCHAR_CAST(c)
#define ZEROMEM(pb,cb) memset(pb,0, cb)
#define MEMCPY(pbTo,pbFrom,cb)  memcpy(pbTo,pbFrom,cb)
#define CREATE_DRM_STRING(g_rgwchCDMDrmStoreName)  DRM_CREATE_DRM_STRING(g_rgwchCDMDrmStoreName)
#define EMPTY_DRM_STRING DRM_EMPTY_DRM_STRING
#define SIZEOF(X) sizeof(X)
#endif

Then the only #ifdef PR_3.3 parts after that would be where the actual number of parameters to a porting kit function differ between releases (like your initial patch).

Reply:
@petegriffin Initially I had the same in mind, but this way we don't need to remove any defines even after we default to v3.3 and the code will be already v3.3 ready.

Reply:
@psivasubramanian I suppose it depends how many porting kit versions this plugin needs to support. If a whole bunch of different porting kit versions it could get a bit messy. I'll let @wouterlucas etc comment on how they would prefer it to be done. I agree with how it's done currently it will be clearer what needs to be removed when PR_3_3 is becomes the default.

Reply:
@petegriffin ok, let @wouterlucas comment on this.

Reply:
It's going to be dependent on the size of the changes. Minor signature changes are fine and as long as we collectively deem it "reasonable" we can keep support multiple versions in 1 module. However if the delta's are too large or the amount of guards become unmanageable we're better off creating seperate OCDMI modules for different versions. For now, with just the two versions we have, this should be fine.